tools/@aws-cdk/spec2cdk/lib/cdk/resource-decider.ts

```diff
@@ -58,39 +61,73 @@ export class ResourceDecider {
     // has both `Arn` and `ResourceArn`, and we want to select the `Arn` property.
     const possibleArnNames = ['Arn', 'ResourceArn', `${this.resource.name}Arn`];
     for (const arn of possibleArnNames) {
-      const att = this.classAttributeProperties.filter((a) => a.propertySpec.name === attributePropertyName(arn));
-      const prop = this.propsProperties.filter((p) => p.propertySpec.name === propertyNameFromCloudFormation(arn));
-      if (att.length > 0 || prop.length > 0) {
-        return att[0] ? att[0].propertySpec : prop[0].propertySpec;
-      }
+      const att = this.classAttributeProperties.find((a) => a.propertySpec.name === attributePropertyName(arn));
+      if (att) { return att.propertySpec; }
+      // TODO: double check -- if an arn is a property, not an attribute, and not a primaryIdentifier
```

Comment:
need a double check here for what to do when the `arn` is from the `property` bag and not a `primaryIdentifier`. The problem is that this is currently generated as `this.arn = props.arn` and based on the spec it doesn't feel like the arn is actually returned via cfn.